src/main/java/thrift/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package thrift.logic.parser;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import thrift.commons.core.Messages;
+import thrift.logic.commands.ListCommand;
+import thrift.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_MONTH ,CliSyntax.PREFIX_TAG);
+        Optional<String> monthArg = argMultimap.getValue(CliSyntax.PREFIX_MONTH);
+        Optional<String> tagArg = argMultimap.getValue(CliSyntax.PREFIX_TAG);
+        if (args.isEmpty()) {
```

Comment:
You can use `!argMultimap.getPreamble().isEmpty()` to check for empty parameters given a correct command.

Reply:
this would cause my next few conditions not to activate though. Like if that path runs for both "list" "list m/mmm"

Reply:
It's alright then, we will see what can be done later on!